android/src/main/java/com/iterable/reactnative/RNIterableAPIModule.java

```diff
@@ -422,9 +436,29 @@ public boolean handleIterableURL(@NonNull Uri uri, @NonNull IterableActionContex
         return false;
     }
 
+    @Override
+    public String onAuthTokenRequested() {
+        IterableLogger.printInfo();
+
+        try {
+            authHandlerCallbackLatch = new CountDownLatch(1);
+            sendEvent(EventName.handleAuthTokenRequestedCalled.name(), null);
+            authHandlerCallbackLatch.await(30, TimeUnit.SECONDS);
```

Comment:
@roninopf It looks like we don't actually call `.countDown()` on the latch anywhere, so it'll always just wait for 30 seconds.